File: backend/Kurator.Core/Entities/Block.cs
using Kurator.Core.Enums;

namespace Kurator.Core.Entities;

/// <summary>
/// Блок (сектор) системы
/// </summary>
public class Block
{
    public int Id { get; set; }

    /// <summary>
    /// Название блока (например: "ОП", "Медиа")
    /// </summary>
    public string Name { get; set; } = string.Empty;

    /// <summary>
    /// Код блока для генерации ID контактов (например: OP, MEDIA)
    /// </summary>
    public string Code { get; set; } = string.Empty;

    /// <summary>
    /// Описание блока (опционально)
    /// </summary>
    public string? Description { get; set; }

    /// <summary>
    /// Статус блока (активный или архивный)
    /// </summary>
    public BlockStatus Status { get; set; } = BlockStatus.Active;

    /// <summary>
    /// Дата создания
    /// </summary>
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    /// <summary>
    /// Дата последнего обновления
    /// </summary>
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties
    /// <summary>
    /// Связь с кураторами через таблицу BlockCurator
    /// </summary>
    public ICollection<BlockCurator> CuratorAssignments { get; set; } = new List<BlockCurator>();

    public ICollection<Contact> Contacts { get; set; } = new List<Contact>();
}

File: frontend/lib/schemas/block.ts
/**
 * Block Zod schemas
 */
import { z } from 'zod';

// Block status enum
export const blockStatusSchema = z.enum(['Active', 'Archived']);
export type BlockStatus = z.infer<typeof blockStatusSchema>;

// Block curator schema
export const blockCuratorSchema = z.object({
  userId: z.number().int().positive(),
  userLogin: z.string(),
  curatorType: z.string(),
  assignedAt: z.string(),
});

export type BlockCurator = z.infer<typeof blockCuratorSchema>;

// Block schema
export const blockSchema = z.object({
  id: z.number().int().positive(),
  name: z.string(),
  description: z.string().nullable().optional(),
  code: z.string(),
  status: blockStatusSchema.or(z.string()),
  curators: z.array(blockCuratorSchema).optional(),
  createdAt: z.string(),
  updatedAt: z.string(),
  // Legacy fields for backwards compatibility
  primaryCuratorId: z.number().optional(),
  backupCuratorId: z.number().optional(),
  primaryCurator: z.any().optional(),
  backupCurator: z.any().optional(),
});

export type Block = z.infer<typeof blockSchema>;

// Block list response (array)
export const blockListResponseSchema = z.array(blockSchema);

export type BlockListResponse = z.infer<typeof blockListResponseSchema>;

// Create block request
export const createBlockRequestSchema = z.object({
  name: z.string().min(1, 'Name is required'),
  code: z.string().min(1, 'Code is required'),
  description: z.string().optional(),
  status: blockStatusSchema.or(z.string()),
});

export type CreateBlockRequest = z.infer<typeof createBlockRequestSchema>;

// Update block request
export const updateBlockRequestSchema = z.object({
  name: z.string().min(1, 'Name is required'),
  description: z.string().optional(),
  status: blockStatusSchema.or(z.string()),
});

export type UpdateBlockRequest = z.infer<typeof updateBlockRequestSchema>;

// Assign curator request
export const assignCuratorRequestSchema = z.object({
  userId: z.number().int().positive(),
  curatorType: z.string(),
});

export type AssignCuratorRequest = z.infer<typeof assignCuratorRequestSchema>;
